### PR TITLE
Doc enhancements for on-prem deployments

### PR DESCRIPTION
**Pull request by LarsAC**
Some additional enhancements which might have helped me to get started.

### CODE REVIEW

**Comment by JoeyZwicker** [comment]
Thanks @LarsAC. We'll review this asap.

File: doc/docs/1.11.x/deploy-manage/deploy/deploy_custom/deploy_custom_pachyderm_deployment_manifest.md [hunk 1]
```diff
@@ -74,4 +74,4 @@ The manifest includes the following sections:
...
+| `Secret`      | Pachyderm uses the Kubernetes `Secret` manifest to store the credentials that <br> are necessary to access object storage. The final manifest uses the <br> command-line arguments that you submit to the `pachctl deploy` <br> command to store such parameters as region, secret, token, and endpoint, that are <br>used to access an object store. The exact values in the secret <br> depend on the kind of object store you configure for your deployment. You <br>can update the values after the deployment either by using `kubectl` <br>to deploy a new `Secret` or the `pachctl deploy storage` command. If you want to modify the secret contents directly, make sure you understand that also on-premise deployments of non-cloud object stores (e.g. MinIO or ECS) make use of the values prefixed with `amazon-` and not the ones prefixed with `minio-`.|
```
Comment (JimmyWhitaker) [inline doc/docs/1.11.x/deploy-manage/deploy/deploy_custom/deploy_custom_pachyderm_deployment_manifest.md:77]:
```suggestion
| `Secret`      | Pachyderm uses the Kubernetes `Secret` manifest to store the credentials that <br> are necessary to access object storage. The final manifest uses the <br> command-line arguments that you submit to the `pachctl deploy` <br> command to store such parameters as region, secret, token, and endpoint, that are <br>used to access an object store. The exact values in the secret <br> depend on the kind of object store you configure for your deployment. You <br>can update the values after the deployment either by using `kubectl` <br>to deploy a new `Secret` or the `pachctl deploy storage` command. Note: If you want to modify the contents of the secret directly, make sure you understand that on-premise deployments of non-cloud object stores (e.g. MinIO or ECS) also use the values prefixed with `amazon-` and not the ones prefixed with `minio-`.|
```

File: doc/docs/1.11.x/deploy-manage/deploy/on_premises.md [hunk 2]
```diff
@@ -165,7 +165,8 @@ We're prefixing each item with how we'll refer to it further on.
 
 1. `OS_ENDPOINT`: The access endpoint.
    For example, MinIO's endpoints are usually something like `minio-server:9000`. 
-   Don't begin it with the protocol; it's an endpoint, not an url.
+   Don't begin it with the protocol; it's an endpoint, not an url. Also, check if your local instance of e.g. MinIO is using SSL/TLS
```
Comment (JimmyWhitaker) [inline doc/docs/1.11.x/deploy-manage/deploy/on_premises.md:168]: By local instance, are you referring to the client running on the local machine?

File: doc/docs/1.11.x/deploy-manage/deploy/on_premises.md [hunk 3]
```diff
@@ -144,7 +144,7 @@ The object store you use must be accessible via a low-latency, high-bandwidth co
 
 For an on-premises deployment, 
 it's not advisable to use a cloud-based storage mechanism.
-Don't deploy an on-premises Pachyderm cluster against cloud-based object stores such as S3 from [AWS](amazon_web_services/index.md), GCS from [Google Cloud Platform](google_cloud_platform.md), Azure Blob Storage from [Azure](azure.md).
+Don't deploy an on-premises Pachyderm cluster against cloud-based object stores such as S3 from [AWS](amazon_web_services/index.md), GCS from [Google Cloud Platform](google_cloud_platform.md), Azure Blob Storage from [Azure](azure.md). Don't get confused that the command line parameters for the object store (`--object-store`) are specifying `s3` - in this case, the argument is referring to the S3 protocol (which is also used by solutions such as MinIO and the likes) and not the Amazon product with the same name.
```
Comment (JimmyWhitaker) [inline doc/docs/1.11.x/deploy-manage/deploy/on_premises.md:147]:
```suggestion
Don't deploy an on-premises Pachyderm cluster against cloud-based object stores such as S3 from [AWS](amazon_web_services/index.md), GCS from [Google Cloud Platform](google_cloud_platform.md), Azure Blob Storage from [Azure](azure.md). Note that the command line parameters for the object store (`--object-store`) are specifying `s3` in reference to the S3 protocol (which is used by solutions such as MinIO and the like) and not the Amazon product with the same name.
```

File: doc/docs/1.11.x/deploy-manage/deploy/on_premises.md (hunk 2 above)
Comment (LarsAC) [reply doc/docs/1.11.x/deploy-manage/deploy/on_premises.md:168]: "Local" is of course not very precise, well spotted. I was referring to the on-prem deployment of MinIO in the k8s cluster. A better formulation might be "Also, check if the object storage deployment (e.g. MinIO) in your cluster uses SSL/TLS ..."

File: doc/docs/1.11.x/deploy-manage/deploy/on_premises.md [hunk 4]
```diff
@@ -165,7 +165,8 @@ We're prefixing each item with how we'll refer to it further on.
 
 1. `OS_ENDPOINT`: The access endpoint.
    For example, MinIO's endpoints are usually something like `minio-server:9000`. 
-   Don't begin it with the protocol; it's an endpoint, not an url.
+   Don't begin it with the protocol; it's an endpoint, not an url. Also, check if your local instance of e.g. MinIO is using SSL/TLS
+   and consider using `--disable-ssl` if not.
```
Comment (JimmyWhitaker) [inline doc/docs/1.11.x/deploy-manage/deploy/on_premises.md:169]:
```suggestion
   Don't begin it with the protocol; it's an endpoint, not an url. Also, check if your object store (e.g. MinIO) is using SSL/TLS.
   If not, disable it using `--disable-ssl`.
```

Comment (JimmyWhitaker) [reply doc/docs/1.11.x/deploy-manage/deploy/on_premises.md:169]: Does this clear it up a bit or did I make it more complicated? hah

**Pull request merged**